stock_portfolio/stock_portfolio/views/default.py

```diff
@@ -46,49 +45,77 @@ def get_auth_view(request):
 
 @view_config(route_name='stock', renderer='../templates/stock-add.jinja2')
 def get_stock_add_view(request):
-    # This code below is not run until form is submitted
     
     if request.method == 'GET':
         try:
-            # print(request)
             symbol = request.GET['symbol']
 
         except KeyError:
             return {}
 
-        response = requests.get(API_URL + '/stock/{}/company'.format(symbol))
-        data = response.json()
-        # print()
-        # print(data)
-        return {'company': data}
-        
-    # else:
-    #     raise HTTPNotFound()
+        try:
+            response = requests.get(API_URL + '/stock/{}/company'.format(symbol))
+            data = response.json()
+            return {'company': data}
+        except ValueError:
+            print('That stock does not exist')
+            return HTTPFound(location=request.route_url('stock'))
         
 
 @view_config(route_name='portfolio', renderer='../templates/portfolio.jinja2')
 def get_portfolio_view(request):
     
     if request.method == 'GET':
-        return {
-            'stocks': MOCK_DATA
-        }
+        try:
+            query = request.dbsession.query(Stock)
+            all_stocks = query.all()
+        except DBAPIError:
+            return DBAPIError(DB_ERR_MSG, content_type='text/plain', status=500)
+        
+        return {'stocks': all_stocks}
 
+    # Get info from API
     if request.method == 'POST':
         symbol = request.POST['symbol']
         response = requests.get(API_URL + '/stock/{}/company'.format(symbol))
         data = response.json()
-        MOCK_DATA.append(data)
-        return {'stocks': MOCK_DATA}
+
+        # Check if info already DB
+        my_object = request.dbsession.query(Stock).filter(Stock.symbol == data['symbol']).first()
+        if not my_object:
+            e = Stock(**data)
+            request.dbsession.add(e)
```

Comment:
This `Stock` should be added to the database from a `POST` to the `/portfolio` route. The spec states that a `POST` to the `/stock` route is what should add a `Stock` to the database. Then the user is redirected to `/portfolio`.